### PR TITLE
Pull Request for Issue2032: Add view for detector stage motor

### PR DESCRIPTION
**Pull request by helfrij**
- Made virtual getter for the detector stage motor in BioXASBeamline.
- Added view to BioXASAppController

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you take a look?

File: source/application/BioXAS/BioXASAppController.cpp [hunk 1]
```diff
@@ -251,6 +251,7 @@ void BioXASAppController::setupUserInterface()
 	addComponentView(BioXASBeamline::bioXAS()->standardsWheel(), "Standards Wheel");
 	addComponentView(BioXASBeamline::bioXAS()->cryostatStage(), "Cryostat Stage");
 	addComponentView(BioXASBeamline::bioXAS()->filterFlipper(), "Filter Flipper");
+	addComponentView(BioXASBeamline::bioXAS()->detectorStageLateral(), "Ge 32-el Stage");
```
Comment (XuetaoLiu-CLS) [inline source/application/BioXAS/BioXASAppController.cpp]: is `detectorStageLateral_` initialized for BioXASMainBeamline? If not, we might need to add a protection here.

**Comment by XuetaoLiu-CLS** [comment]
Looks good to me.

Time Estimates: 5 minutes

File: source/application/BioXAS/BioXASAppController.cpp (hunk 1 above)
Comment (dretrex) [inline source/application/BioXAS/BioXASAppController.cpp]: @XuetaoLiu-CLS, I believe that protection lives inside of the `addComponentView()`.

**Comment by dretrex** [comment]
Looks good.
Time Estimate: 2 minutes

File: source/application/BioXAS/BioXASAppController.cpp (hunk 1 above)
Comment (helfrij) [inline source/application/BioXAS/BioXASAppController.cpp]: Good eye, @XuetaoLiu-CLS! By default, the getters defined in BioXASBeamline always return 0, unless it's for a component that's created in that class. For Main, this call would return 0. This condition is checked in createComponentView().

**Pull request merged**